### PR TITLE
CI: save compose logs and metadata

### PR DESCRIPTION
**Pull request by achilleas-k**
This pull request includes:

- [ ] adequate testing for the new functionality or fixed issue
- [ ] adequate documentation informing people about the change such as
  - [ ] create a file in [news/unreleased](https://github.com/osbuild/osbuild-composer/tree/main/docs/news/unreleased) directory if this change should be mentioned in the release news
  - [ ] submit a PR for the [guides](https://github.com/osbuild/guides) repository if this PR changed any behavior described there: https://www.osbuild.org/guides/

Defined a general purpose artifact save directory where test scripts can add files to be saved for inspecting and troubleshooting builds in CI.
I modified the ostree test scripts to save their compose manifest and logs.

### CODE REVIEW

File: test/cases/ostree-ng.sh [hunk 1]
```diff
@@ -297,6 +297,14 @@ build_image container "${CONTAINER_TYPE}"
 greenprint "📥 Downloading the container image"
 sudo composer-cli compose image "${COMPOSE_ID}" > /dev/null
 
+# Save the build manifest and logs to the artifacts directory
+greenprint "Saving build metadata and logs"
```
Comment (henrywang) [inline test/cases/ostree-ng.sh:301]: @achilleas-k We have fetch build log and metadata on function `get_compose_log` and `get_compose_metadata` already. But we need archive them.

Comment (achilleas-k) [reply test/cases/ostree-ng.sh]: Perfect thank you.  I've updated the commit to use those functions.

File: test/cases/ostree-ng.sh [hunk 2]
```diff
@@ -72,6 +72,7 @@ STAGE_REPO_ADDRESS=192.168.200.1
 STAGE_REPO_URL="http://${STAGE_REPO_ADDRESS}/repo/"
 QUAY_REPO_URL="docker://quay.io/osbuild/testing-rhel-edge-push"
 QUAY_REPO_TAG=$(tr -dc a-z0-9 < /dev/urandom | head -c 4 ; echo '')
+ARTIFACTS="ci-artifacts"
```
Comment (ondrejbudai) [inline test/cases/ostree-ng.sh:75]: let's maybe `mkdir` here so we don't have to do it twice? The same applies to the other script.

**Review by ondrejbudai — APPROVED** [review]
perfection

**Review by atodorov — APPROVED** [review]
LGTM

**Pull request merged**